### PR TITLE
Added `AssetPublicQRViewSet` to handle QR code based asset retrieval

### PR DESCRIPTION
**Pull request by Ashesh3**
This PR introduces a new viewset, `AssetPublicQRViewSet`, which allows the retrieval of assets using their QR code identifiers. The viewset checks if the provided identifier is a UUID or the primary key of the asset. If the identifier is not found in the cache, the asset is retrieved from the database, serialized, and stored in the cache for future requests. This update also includes the registration of the new viewset in the API router under the public endpoints.

Required for: https://github.com/coronasafe/care_fe/issues/3480

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

**Comment by sainak** [comment]
@Ashesh3 add tests for the retrieve method

**Review by sainak — COMMENTED** [review]
needs test

**Review by sainak — APPROVED** [review]
LGTM

File: care/facility/api/viewsets/asset.py [hunk 1]
```diff
@@ -174,6 +176,36 @@ def retrieve(self, request, *args, **kwargs):
...
+
+        key = "asset:qr:" + kwargs["qr_code_id"]
+        hit = cache.get(key)
+        if not hit:
+            if is_uuid:
+                instance = self.get_object()
+            else:
+                instance = Asset.objects.filter(pk=kwargs["qr_code_id"]).first()
+                if not instance:
+                    return Response(status=status.HTTP_404_NOT_FOUND)
```
[the remark this answers is not in the document]
Comment (Ashesh3) [reply care/facility/api/viewsets/asset.py]: Done!

**Comment by vigneshhari** [comment]
You also need to create a signal to delete the cache when an asset updates, need tests for that edge as well.

**Comment by Ashesh3** [comment]
> You also need to create a signal to delete the cache when an asset updates, need tests for that edge as well.

@vigneshhari added, could you review?

**Review by sainak — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]
